File: WorldSpawnAPI/src/main/java/me/ifydev/worldspawn/database/ConnectionInformation.java
package me.ifydev.worldspawn.database;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Map;

/**
 * @author Innectic
 * @since 08/06/2018
 */
@AllArgsConstructor
public class ConnectionInformation {
    @Getter private String url;
    @Getter private String database;
    @Getter private int port;
    @Getter private String username;
    @Getter private String password;
    @Getter private Map<String, Object> meta;
}

File: API/src/main/java/me/ifydev/worldspawn/api/backend/ConnectionInformation.kt
package me.ifydev.worldspawn.api.backend

/**
 * @author Innectic
 * @since 08/07/2018
 *
 * Information used to connect to the data storage backend.
 */
data class ConnectionInformation(
        val host: String, val database: String, val port: Int, val username: String, val password: String,
        val meta: Map<String, Any>, val type: Backend.Types)